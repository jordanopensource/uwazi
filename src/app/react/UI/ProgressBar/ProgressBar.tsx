import React from 'react';

type progressBarProps = {
  max?: number;
  value?: number;
  useProgressColors?: boolean;
  showNumericValue?: boolean;
};

const ProgressBar = ({
  max = 100,
  value = 0,
  useProgressColors = false,
  showNumericValue = true,
}: progressBarProps) => {
  const percentage = (100 * value) / max;
  let progressColor;

  switch (true) {
    case percentage < 40:
      progressColor = '#89E79E';
      break;

    case percentage > 80:
      progressColor = '#B24138';
      break;

    default:
      progressColor = '#F9CA70';
      break;
  }

  return (
    <div className="uw-progress-bar--container">
      <div className="uw-progress-bar">
        <div
          className="uw-progress-bar--progress"
          style={{
            width: `${percentage}%`,
            ...(useProgressColors && { backgroundColor: progressColor }),
          }}
        />
      </div>
      {showNumericValue && (
        <div className="uw-progress-bar--counter">
          {value}/{max}
        </div>
      )}
    </div>
  );
};

export default ProgressBar;
