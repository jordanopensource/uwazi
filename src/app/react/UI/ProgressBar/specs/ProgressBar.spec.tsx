import React from 'react';
import { shallow } from 'enzyme';

import ProgressBar from '../ProgressBar';

describe('ProgressBar', () => {
  it('should render progress bar as value percentage of max', () => {
    const component = shallow(<ProgressBar value={200} max={250} />);
    expect(component).toMatchSnapshot();
  });

  describe('colors', () => {
    it('should show colors when enabled based on percentage', () => {
      const component = shallow(<ProgressBar value={3} max={10} useProgressColors />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('numeric value', () => {
    it('should show it by default', () => {
      const component = shallow(<ProgressBar value={3} max={10} />);
      expect(component).toMatchSnapshot();
    });
  });
});
