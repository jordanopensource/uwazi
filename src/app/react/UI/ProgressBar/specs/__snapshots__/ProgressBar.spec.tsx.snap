// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar colors should show colors when enabled based on percentage 1`] = `
<div
  className="uw-progress-bar--container"
>
  <div
    className="uw-progress-bar"
  >
    <div
      className="uw-progress-bar--progress"
      style={
        Object {
          "backgroundColor": "#89E79E",
          "width": "30%",
        }
      }
    />
  </div>
  <div
    className="uw-progress-bar--counter"
  >
    3
    /
    10
  </div>
</div>
`;

exports[`ProgressBar numeric value should show it by default 1`] = `
<div
  className="uw-progress-bar--container"
>
  <div
    className="uw-progress-bar"
  >
    <div
      className="uw-progress-bar--progress"
      style={
        Object {
          "width": "30%",
        }
      }
    />
  </div>
  <div
    className="uw-progress-bar--counter"
  >
    3
    /
    10
  </div>
</div>
`;

exports[`ProgressBar should render progress bar as value percentage of max 1`] = `
<div
  className="uw-progress-bar--container"
>
  <div
    className="uw-progress-bar"
  >
    <div
      className="uw-progress-bar--progress"
      style={
        Object {
          "width": "80%",
        }
      }
    />
  </div>
  <div
    className="uw-progress-bar--counter"
  >
    200
    /
    250
  </div>
</div>
`;
